src/sample/direct.jl

```diff
@@ -122,3 +122,11 @@ end
 function Base.length(dc::DirectCall)
     return length(dc.prefix_tree)
 end
+
+function Base.haskey(dc::DirectCall{K,T,P}, clock) where {K,T,P}
+    if clock isa K
+        haskey(dc.prefix_tree.index, clock)
```

Comment:
For the Direct method, do you want to check whether the rate is 0.0 or not? What are the semantics you're going for?